File: hotel-web/src/api/payment.ts
import request from './request'
import type { ApiResult, PaymentResponse } from '@/types'

export function createPayment(data: {
  orderId: string
  orderNo: string
  hotelId: string
  memberId?: string
  amount: number
  paymentType?: number
  payChannel?: string
  businessType?: string
}): Promise<ApiResult<PaymentResponse>> {
  return request.post('/pay/create', data).then(r => r.data)
}

export function callbackPayment(data: {
  paymentNo: string
  tradeNo: string
  status: number
  message?: string
}): Promise<ApiResult<PaymentResponse>> {
  return request.post('/pay/callback', data).then(r => r.data)
}

export function getPayment(paymentId: string): Promise<ApiResult<PaymentResponse>> {
  return request.get(`/pay/query/${paymentId}`).then(r => r.data)
}

export function getPaymentByOrder(orderId: string): Promise<ApiResult<PaymentResponse>> {
  return request.get(`/pay/queryByOrder/${orderId}`).then(r => r.data)
}

File: hotel-services/hotel-payment/src/main/java/com/hotel/payment/model/entity/Payment.java
package com.hotel.payment.model.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("pay_payment")
public class Payment {
    @TableId
    private String paymentId;
    private String paymentNo;
    private String orderId;
    private String orderNo;
    private String hotelId;
    private String memberId;
    private Integer businessType;
    private String businessId;
    private Integer paymentType;
    private String payChannel;
    private BigDecimal amount;
    private String currency;
    private Integer status;
    private String tradeNo;
    private LocalDateTime payTime;
    private LocalDateTime expireTime;
    private String refundId;
    private BigDecimal refundAmount;
    private LocalDateTime refundTime;
    private String refundReason;
    private String clientIp;
    private String paymentParams;
    private LocalDateTime createdTime;
    private LocalDateTime updatedTime;
}
